24 drop dict caching

routesix opened the pull request:
Ability to use a dictionary as the IP cache has been removed.

routesix committed:
Dropped Dict Caching

Went through the code to remove artifacts of dictionary caching. The one portion that remains is to remove the CACHE_TYPE class from const.py

routesix committed:
Dropped CACHE_TYPE Class

Removed the CACHE_TYPE class - the last vestige of dictionary caching.

routesix committed:
Update test_server.py

Removed some tests for dictionary caching.

b0urb0n commented on delirium/dns/fakednsserver.py:
```diff
@@ -8,7 +8,7 @@
+                 ip_range=DEFAULT_ADDR_RANGE, cache_path=DEFAULT_DB_PATH):
         self._addr = addr
         self._port = port
 
```
We discussed dropping the `database.create_cache` call and just instantiating an instance of  `database.CacheDatabase` directly.

b0urb0n commented on delirium/dns/fakeresolver.py:
```diff
@@ -1,6 +1,4 @@
 from dnslib.server import BaseResolver
-#from dnslib.dns import RCODE
-#from dnslib import QTYPE
 from dnslib import *
```
non-internal `import *` statements are generally considered bad practice.  Was more than just `dns.RCODE` and `QTYPE` needed from `dnslib`?

b0urb0n commented:
This is a great start to this task.  In addition to some minor comments above.  I'd like you to consider dropping the `delirium.dns.models` package and replacing it with a `delirium.dns.cache` module (single `.py` file).  As we've removed a bunch of unused code, there's no longer a need for this component to have its own package.

routesix committed:
Cache.py own Module

Made cache.py its own module

routesix committed:
Revert "Cache.py own Module"

This reverts commit e29445d99f9990cb4d9cb6e14fa8166f1de745c8.

routesix committed:
More Precise Imports

No longer load all of dnslib. Take only RCODE and QTYPE

routesix committed:
Replaced Models with Cache

Consoldiated database and cache modules and eliminated Models package

b0urb0n commented on delirium/dns/cache.py:
```diff
@@ -25,18 +28,73 @@
+    for i in itertools.cycle(range(start, end + 1)):
+        yield i
+
+class CacheObject:
```
I'd like to see this abstract class and `CacheDatabase` combined into one class.  As we're only supporting one cache-type now, there really isn't a need for an abstract class.  I'll explain why on our call tonight.

routesix committed:
Removed Abstract Class

I removed the abstract cache class from cache.py. The testing fails, but I attribute that to the way the tests were set up. I recommend fixing the tests as part of #15

routesix committed:
Added Decorators

Previous errors were not due to testing. The issue was a lack of the appropriate decorators in the CacheDatabase class.

b0urb0n commented on delirium/dns/cache.py:
```diff
@@ -25,26 +27,76 @@
+        return struct.unpack('!L', socket.inet_aton(value))[0]
+
+    @staticmethod
+    def add_record(self, label):
```
Duplicate method below

b0urb0n commented on delirium/dns/cache.py:
```diff
@@ -25,26 +27,76 @@
+        pass
+
+    @staticmethod
+    def close(self):
```
Duplicate method below

b0urb0n commented on delirium/dns/cache.py:
```diff
@@ -25,26 +27,76 @@
+        pass
+
+    @staticmethod
+    def get_addr_by_name(self, value):
```
Duplicate method below

b0urb0n commented on delirium/dns/cache.py:
```diff
@@ -25,26 +27,76 @@
+        pass
+
+    @staticmethod
+    def get_name_by_addr(self, value):
```
Duplicate method below

b0urb0n commented on delirium/dns/cache.py:
```diff
@@ -25,26 +27,76 @@
+        pass
+
+    @staticmethod
+    def prune_stale(self):
```
Duplicate method below

routesix committed:
Removed Stub Functions

Removed a number of stub functions left over from the abstract class merger.
